Sources/ClientRuntime/Networking/Http/HttpTelemetry.swift

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import protocol SmithyHTTPAPI.HTTPClient
+import struct Smithy.Attributes
+import struct Smithy.AttributeKey
+
+/// Container for HTTPClient telemetry, including configurable attributes and names.
+///
+/// Note: This is intended to be used within generated code, not directly.
+public class HttpTelemetry {
```

Comment:
`HttpTelemetry ` is a container for all the state (e.g. attributes) and meters (used for metrics) for a given `HTTPClient` instance. The container does not emit any metrics itself, and only provides convenient access standardized metrics.

Metrics captured:

- smithy.client.http.connections.acquire_duration
- smithy.client.http.connections.limit
- smithy.client.http.connections.usage
- smithy.client.http.connections.uptime
- smithy.client.http.requests.usage
- smithy.client.http.requests.queued_duration
- smithy.client.http.bytes_sent
- smithy.client.http.bytes_received